src/components/views/settings/tabs/user/EncryptionUserSettingsTab.tsx

```diff
@@ -54,71 +61,115 @@ interface EncryptionUserSettingsTabProps {
     initialState?: State;
 }
 
+const useKeyBackupIsEnabled = (): boolean | undefined => {
+    const [isEnabled, setIsEnabled] = useState<boolean | undefined>(undefined);
+    const [loading, setLoading] = useState(true);
+
+    const matrixClient = useMatrixClientContext();
+
+    const checkStatus = useCallback(async () => {
+        const crypto = matrixClient.getCrypto()!;
+        const info = await crypto.getKeyBackupInfo();
+        setIsEnabled(Boolean(info?.version));
+    }, [matrixClient, setIsEnabled]);
+
+    useEffect(() => {
+        (async () => {
+            await checkStatus();
+            setLoading(false);
+        })();
+    }, [checkStatus, setLoading]);
+
+    useEventEmitter(matrixClient, ClientEvent.AccountData, (event: MatrixEvent): void => {
+        const type = event.getType();
+        // Recheck the status if this account data has been updated as this implies it has changed
+        if (type === "m.org.matrix.custom.backup_disabled") {
+            checkStatus();
+        }
+    });
+
+    return loading ? undefined : isEnabled;
+};
+
 /**
  * The encryption settings tab.
  */
 export function EncryptionUserSettingsTab({ initialState = "loading" }: EncryptionUserSettingsTabProps): JSX.Element {
     const [state, setState] = useState<State>(initialState);
 
     const checkEncryptionState = useCheckEncryptionState(state, setState);
+    const keyBackupIsEnabled = useKeyBackupIsEnabled();
 
     let content: JSX.Element;
-    switch (state) {
-        case "loading":
-            content = <InlineSpinner aria-label={_t("common|loading")} />;
-            break;
-        case "set_up_encryption":
-            content = <SetUpEncryptionPanel onFinish={checkEncryptionState} />;
-            break;
-        case "secrets_not_cached":
-            content = (
-                <RecoveryPanelOutOfSync
-                    onFinish={checkEncryptionState}
-                    onForgotRecoveryKey={() => setState("reset_identity_forgot")}
-                />
-            );
-            break;
-        case "main":
-            content = (
-                <>
-                    <RecoveryPanel
-                        onChangeRecoveryKeyClick={(setupNewKey) =>
-                            setupNewKey ? setState("set_recovery_key") : setState("change_recovery_key")
-                        }
+    if (keyBackupIsEnabled === undefined) {
+        content = <Spinner />;
+    } else {
+        switch (state) {
+            case "loading":
+                content = <InlineSpinner aria-label={_t("common|loading")} />;
+                break;
```

Comment:
It looks like we have two loading state, I'll be happy if we can keep one loading state